boltdesignsystem/bolt
# Upgrade Bolt from Preact v8.x to Preact v10

## Jira
N/A

## Summary
Upgrades Bolt from Preact v8.x to Preact v10.

## Details
These updates are ported over from the larger #1542; this PR includes the work mentioned in the Preact v8.x [upgrade guide](https://preactjs.com/guide/v10/upgrade-guide) in addition to some minor updates to our Preact-based web component renderer + a small number of web components still rendering using Preact.

Also worth pointing out, this PR also requires us to upgrade Pattern Lab's UI to Preact v10 due to how Yarn dependencies in the monorepo are deduplicated and shared.

## How to test
- Review changes
- Confirm tests passing as expected
- Do a gut check on the Bolt components that are still rendering with Preact
  - Block List (small change required)
  - Code Snippet (small change required)
  - Video (no changes required)
  - Dropdown (no changes required)
  - Typeahead  (no changes required)
  - Icon (no changes required)
- Confirm the Pattern Lab UI (especially the Tooltips that show up when you hover over the viewport resizer icons) are displaying correctly
